server/middleware/journey/redirectCheckAnswers.ts

```diff
@@ -0,0 +1,68 @@
+import { NextFunction, Request, Response } from 'express'
+import { FLASH_KEY__VALIDATION_ERRORS } from '../setUpFlash'
+
+/**
+ * Wraps res.redirect and res.render to redirect to check-your-answers page if
+ * check answers page has been reached in the current journey and there are no validation errors.
+ * @param excludePaths An array of regexs for this middleware to not activate on
+ */
+export default function redirectCheckAnswersMiddleware(excludePaths: RegExp[] = []) {
+  return (req: Request, res: Response, next: NextFunction): void => {
+    if (!req.originalUrl || excludePaths.some(itm => req.originalUrl.match(itm))) {
+      return next()
+    }
+
+    const subPaths = req.originalUrl.split('/')
+    const journeyIdIndex = subPaths.findIndex(itm =>
+      itm.match(/[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/),
+    )
+
+    if (journeyIdIndex === -1) {
+      return next()
+    }
+
+    const journeyId = subPaths[journeyIdIndex]
+    const checkAnswersUrl = [...subPaths.slice(0, journeyIdIndex), journeyId, `check-your-answers`].join('/')
+
+    const resRender = res.render
+    res.render = (view: string, options?) => {
+      if (
+        options &&
+        'backUrl' in options &&
+        options.backUrl &&
+        req.session.journeyData?.[journeyId]?.reachedCheckAnswers
+      ) {
+        resRender.call(res, view, { ...options, backUrl: checkAnswersUrl } as never)
+      } else {
+        resRender.call(res, view, options as never)
+      }
+    }
+
```

Comment:
These are the rules we'll need to define in more detail - as some parts of the journey, if changed, will likely affect other parts. If you change the visitType, then any choice of date, time, slot might no longer be valid ... so they'll be some fine-tuning here. Perhaps each step, if changed, nullifies the session data of dependant steps? Something like that - TBD.